lib/BackbeatConsumer.js

```diff
@@ -550,7 +621,13 @@ class BackbeatConsumer extends EventEmitter {
                 // Ignore exceptions if we are not connected
                 const logger = this._consumer.isConnected() ? this._log.error : this._log.debug;
                 logger.bind(this._log)('rdkafka.assign failed', { e: e.toString(), assignment });
+                return;
+            }
+
+            if (this._circuitBreaker.state !== BreakerState.Nominal) {
```

Comment:
instead of having the `return`, this can be added in the `try` block to be more readable